src/AI/AICombatUtils.java

```diff
@@ -66,29 +66,29 @@ public static double scoreAttackAction(Unit unit, GameAction action, GameMap map
   /**
    * @return The area and severity of threat from the unit, against the specified target type
    */
-  public static Map<XYCoord, Double> findThreatPower(GameMap gameMap, Unit unit, UnitModel target)
+  public static Map<XYCoord, Integer> findThreatPower(GameMap gameMap, Unit unit, UnitModel target)
   {
     XYCoord origin = new XYCoord(unit.x, unit.y);
     return findThreatPower(gameMap, unit, origin, target);
   }
-  public static Map<XYCoord, Double> findThreatPower(GameMap gameMap, Unit unit, XYCoord origin, UnitModel target)
+  public static Map<XYCoord, Integer> findThreatPower(GameMap gameMap, Unit unit, XYCoord origin, UnitModel target)
   {
-    Map<XYCoord, Double> shootableTiles = new HashMap<XYCoord, Double>();
+    Map<XYCoord, Integer> shootableTiles = new HashMap<>();
     PathCalcParams pcp = new PathCalcParams(unit, gameMap);
     pcp.start = origin;
     pcp.includeOccupiedSpaces = true; // We assume the enemy knows how to manage positioning within his turn
     ArrayList<Utils.SearchNode> destinations = pcp.findAllPaths();
     for( WeaponModel wep : unit.model.weapons )
     {
-      double damage = (null == target)? 1 : wep.getDamage(target) * unit.getHPFactor();
+      int damage = (null == target)? 1 : wep.getDamage(target) * unit.getHP() / UnitModel.MAXIMUM_HP;
```

Comment:
`getHPFactor` still exists. Why break it out this way?

Reply:
I changed it to [0,10] instead of [0,1], which was probably ill-advised... but made a few things more convenient.
And it banishes floating point from a few calcs, which I see as a win.